execution/aggregate/khashaggregate.go

```diff
@@ -205,41 +209,113 @@ func (a *kAggregate) init(ctx context.Context) error {
 	return nil
 }
 
-func (a *kAggregate) aggregate(t int64, result *[]model.StepVector, k int, sampleIDs []uint64, samples []float64) {
-	for i, sId := range sampleIDs {
-		h := a.inputToHeap[sId]
-		switch {
-		case h.Len() < k:
-			heap.Push(h, &entry{sId: sId, total: samples[i]})
+func (a *kAggregate) aggregate(t int64, result *[]model.StepVector, k int, sampleIDs []uint64, samples []float64, histogramIDs []uint64, histograms []*histogram.FloatHistogram) {
+	groupsRemaining := len(a.heaps)
+
+	if a.aggregation != parser.LIMITK {
+		for i, sId := range sampleIDs { // BOTTOMK, TOPK
```

Comment:
Maybe we could make the code more straightforward by writing this instead of putting it into a comment:

```
if a.aggregation == parser.BOTTOMK || a.aggregation == parser.TOPK {
...
} else {
...
}
```